src/Simulation/Simulation.jl

```diff
@@ -862,13 +826,22 @@ function _calculate_potential!( sim::Simulation{T, CS}, potential_type::UnionAll
                 (T(1e-5), T(1e-5), T(1e-5))
             end
         end
-        if use_nthreads > Base.Threads.nthreads()
-            use_nthreads = Base.Threads.nthreads();
-            @warn "`use_nthreads` was set to `1`. The environment variable `JULIA_NUM_THREADS` must be set appropriately before the julia session is started."
-        end
         refine = !ismissing(refinement_limits)
         if !(refinement_limits isa Vector) refinement_limits = [refinement_limits] end
         n_refinement_steps = length(refinement_limits)
+
+        max_nthreads, guess_nt = if use_nthreads isa Int 
+            if use_nthreads > Base.Threads.nthreads()
+                use_nthreads = Base.Threads.nthreads();
+                @warn "`use_nthreads` was set to `1`. The environment variable `JULIA_NUM_THREADS` must be set appropriately before the julia session is started."
+            end
+            fill(use_nthreads, n_refinement_steps+1), true
+        else
+            _nt = fill(maximum(use_nthreads), n_refinement_steps+1)
+            _nt[1:length(use_nthreads)] = use_nthreads
```

Comment:
What happens if `use_nthreads` is longer than `n_refinement_steps+1`?
This would look like a `BoundsError` to me